src/asm_ghidra.cpp

```diff
@@ -11,14 +11,22 @@ static SleighAsm sasm;
 static int disassemble(RAsm *a, RAsmOp *op, const ut8 *buf, int len)
 {
 	int r = 0;
+	RIO *rio = nullptr;
 	if(!a->cpu)
 		return r;
 #ifndef DEBUG_EXCEPTIONS
 	try
 	{
 #endif
 		RBin *bin = a->binb.bin;
-		sasm.init(a->cpu, bin? bin->iob.io: nullptr, SleighAsm::getConfig(a));
+		if(!bin)
+		{
+			rio = r_io_new();
```

Comment:
@trufae I think RIO here is wrongly created.

Reply:
i dont see why is this needed

Reply:
Because in SLEIGH, it need RIO, or something like it.

Reply:
why? i dont use the instance from bin , instead of creating a dummy instance with no maps loaded or anything setup?  this looks wrong to me. write fill fail below and all the reads will be ffff